underdocs-error/src/main/kotlin/underdocs/error/pretty/DefaultErrorPrettyPrinter.kt

```diff
@@ -1,7 +1,34 @@
 package underdocs.error.pretty
 
+import underdocs.error.BaseUnderdocsError
+
 class DefaultErrorPrettyPrinter : ErrorPrettyPrinter {
     override fun print(exception: Exception): String {
-        TODO("Not yet implemented")
+        var prettyPrintedError = """"""
+
+        if (exception is BaseUnderdocsError) {
+            prettyPrintedError = """
+                |${exception.title.toUpperCase()} - Error Code ${exception.code}
+                |Details
+                |${exception.getDescription()}
+                |
+                |Possible solution
+                |${exception.getSolution()}
+            """.trimMargin()
+
+            if (exception.cause != null) prettyPrintedError += "\nError cause:\n${exception.cause?.message}"
```

Comment:
While this is correct, you could also utilize one of Kotlin's [Scope Functions](https://kotlinlang.org/docs/reference/scope-functions.html#scope-functions). What do you think, would you give them a shot?

Reply:
Thanks for that! This was such a wacky "workaround".

Reply:
It was fine, but just too Java-ish :)